python/ray/data/read_api.py

```diff
@@ -377,6 +378,75 @@ def read_parquet(
     )
 
 
+@PublicAPI(stability="alpha")
+def read_images(
+    root: str, size: Optional[Tuple[int, int]] = None, mode: Optional[str] = None
+):
+    """Read datasets like `ImageNet <https://www.image-net.org/>`_.
+
+    This function works with any directory where images are arranged in this way:
+
+    .. code-block::
+
+        root/dog/xxx.png
+        root/dog/xxy.png
+        root/dog/[...]/xxz.png
+
+        root/cat/123.png
+        root/cat/nsdf3.png
+        root/cat/[...]/asd932_.png
```

Comment:
Please add a sentence that explains our position on extending the intake format of this API